spline_background.py

```diff
@@ -0,0 +1,73 @@
+import numpy as np
+from scipy import interpolate
+
+
+def bspline_background(variable, x_values, dim=None):
+    """
+    Use scipy bspline method to create a fitting function to data in `variable`.
+    Knots and splines are evaluated internally.
+
+    Parameters
+    ----------
+        variable: scipp variable
+            The variable to which the spline should be applied
+        x_values: nupmy array
+            X values for the variable data
+        dim: scipp Dim
+            The dimension along which the spline should be calculated
+    """
+
+    if dim is None:
+        raise ValueError("bspline_background: dimension must be specified.")
+
+    values = variable[dim, :].values
+    errors = variable[dim, :].variances
+    weights = [1.0/i*i for i in errors]
```

Comment:
- Why not operate with th numpy array, avoiding the iteration and list comprehension?
- Are you sure about the square in the denominator? As the name says, `Variable.variances` are the variances, not the standard-deviations. Note that Mantid stores standard-deviations.